### ISSUE TITLE
Roomster cluster speed

### ISSUE THREAD

**Issue by kerrcz** [r00li/CarCluster#24]
I have kinda working roomster cluster. RPM and indicators work but thats all i cant get working anything else any ideas to make atleast speed work?

**Comment by kerrcz** [comment]
i have the wiring (in czech) i have rn just pin 8 and 7. i tried adding to that 5and 6 and the whol ecluster didnt do anything. it was on but i didnt react at all 
![obrázek_2024-11-10_144106128](https://github.com/user-attachments/assets/68b87bba-089e-4a89-9d2e-960cb5731994)

**Comment by r00li** [comment]
Hmmm, looking at the pinout - speed signal is pin 6. That makes it sound like the speed isn't actually provided through CAN, but is instead done via a pin. That would explain some things. Also, I see some other stuff that should be on CAN apparently isn't.

**Comment by kerrcz** [comment]
could i somehow simulate that with esp32?

**Comment by kerrcz** [comment]
also that would explain why when i tried plugging in the speed into canbus the speed was jumping all around the place

**Comment by r00li** [comment]
> could i somehow simulate that with esp32?

Yeah, it shouldn't be that hard. I think you should be able to do it with standard arduino PWM. Increasing the frequency will probably increase the speed. Start with something like 1Hz and go higher.

**Comment by kerrcz** [comment]
and could it be possible to make it work with your code?

**Comment by r00li** [comment]
Yes, it should. But you will of course need to do some modifications.

**Comment by kerrcz** [comment]
if you hade some free time could you make those modifications? since i really dont know how to code

**Comment by r00li** [comment]
Definitely not. Especially because you need the cluster there to actually check what is happening. The PWM stuff should be quite trivial - just search google for "ESP32 arduino PWM" and you will find some tutorials. But you will need to know some arduino basics first.

**Comment by kerrcz** [comment]
it needs two pins so one that sends and ground or?

**Comment by r00li** [comment]
I think the input is for the cluster to show stuff, output is for some other devices in the car that also need that information. So you would only need 1 pin.

**Comment by kerrcz** [comment]
could you atleast try to write some really basic code for arduino nano so  that i can try it? and then i would try to continue by myself

**Comment by r00li** [comment]
Well on an arduino Nano you can just try with tone first:
https://docs.arduino.cc/language-reference/en/functions/advanced-io/tone/

just write `tone(PIN, FREQUENCY);`. Obviously replace PIN with your PIN and FREQUENCY with the frequency that you want. As I said start with 1 and then increase it to 10, 100 and see what happens.

**Comment by kerrcz** [comment]
i tried this and it didnt do anything https://docs.arduino.cc/tutorials/generic/secrets-of-arduino-pwm/
